qcodes_contrib_drivers/drivers/Thorlabs/PM100D.py

```diff
@@ -0,0 +1,186 @@
+            get_parser=lambda val: float(val)/1e9,
+            set_parser=lambda val: float(val)*1e9,
+            vals=vals.Numbers(185e-9, 25e-6),
+            instrument=self
+        )
+
+        self.power = Parameter(
+            'power',
+            label='Measured power',
+            unit='W',
+            get_cmd=self._get_power,
+            vals=vals.Numbers(),
+            instrument=self
+        )
+
+        self.attenuation = Parameter(
+            'attenuation',
+            label='Attenuation',
+            unit='dB',
+            get_cmd='CORR?',
+            get_parser=float,
+            set_cmd='CORR {}',
+            set_parser=float,
+            vals=vals.Numbers(-60, 60),
+            instrument=self
+        )
+
+        self.power_range = Parameter(
+            'power_range',
+            label='Power range',
+            unit='W',
+            get_cmd='SENS:POW:RANG:UPP?',
+            set_cmd='SENS:POW:RANG:UPP {}',
+            get_parser=float,
+            set_parser=float,
+            vals=vals.Numbers(),
+            instrument=self
+        )
+
+        self.auto_range = Parameter(
+            'auto_range',
+            label='Auto range power',
+            get_cmd='SENS:POW:RANG:AUTO?',
+            set_cmd='SENS:POW:RANG:AUTO {}',
+            val_mapping=create_on_off_val_mapping(on_val='1', off_val='0'),
+            vals=vals.Ints(0, 1),
+            instrument=self
+        )
+
+        self.frequency = Parameter(
+            'frequency',
+            unit='Hz',
+            get_cmd='MEAS:FREQ?',
+            get_parser=float,
+            vals=vals.Numbers(),
+            instrument=self
+        )
+
+        self.current = Parameter(
+            'current',
+            label='Current',
+            unit='A',
+            get_cmd='MEAS:CURR?',
+            get_parser=float,
+            vals=vals.Numbers(),
+            instrument=self
+        )
+
+        self.current_range = Parameter(
+            'current_range',
+            label='Current range',
+            unit='A',
+            get_cmd='SENS:CURR:RANG:UPP?',
+            get_parser=float,
+            vals=vals.Numbers(),
+            instrument=self
+        )
+
+        self.zero_value = Parameter(
+            'zero_value',
+            unit='W',
+            get_cmd='CORR:COLL:ZERO:MAGN?',
+            get_parser=float,
+            vals=vals.Numbers(),
+            instrument=self
+        )
+
+        self.beam_diameter = Parameter(
+            'beam_diameter',
+            label='Beam diameter',
+            unit='m',
+            get_cmd='CORR:BEAM?',
+            set_cmd='CORR:BEAM {}',
+            get_parser=lambda val: float(val)/1e3,
+            set_parser=lambda val: float(val)*1e3,
+            vals=vals.Numbers(),
+            instrument=self
+        )
+
+        self.write('STAT:OPER:PTR 512')
+        sleep(.2)
+        self.write('STAT:OPER:NTR 0')
+        sleep(5)
+        self.ask('STAT:OPER?')
```

Comment:
i'd recommend to create methods (or even parameters) for these 3 commands, that way the driver code becomes more readable in that it's easy to see "what is happening" rather than "which commands are being sent" which can be very cryptic